### ISSUE TITLE
Redis cluster config doesn't work

### ISSUE THREAD

**Issue by kimxogus** [getsentry/sentry#19910]
## Important Details

How are you running Sentry?

On-Premise w/ Docker, version x.y.z
Docker tag getsentry/sentry:cc9f7d1

## Description

I configured `is_redis_cluster: true` in config.yml but log shows
```
05:33:42 [WARNING] sentry.utils.geo: settings.GEOIP_PATH_MMDB not configured.
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/__init__.py", line 166, in main
    cli(prog_name=get_prog(), obj={}, max_content_width=100)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/decorators.py", line 68, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/decorators.py", line 29, in inner
    configure()
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/__init__.py", line 129, in configure
    configure(ctx, py, yaml, skip_service_validation)
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/settings.py", line 158, in configure
    skip_service_validation=skip_service_validation,
  File "/usr/local/lib/python2.7/site-packages/sentry/runner/initializer.py", line 310, in initialize_app
    django.setup()
  File "/usr/local/lib/python2.7/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python2.7/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models()
  File "/usr/local/lib/python2.7/site-packages/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/local/lib/python2.7/site-packages/sentry/models/__init__.py", line 5, in <module>
    import_submodules(globals(), __name__, __path__)
  File "/usr/local/lib/python2.7/site-packages/sentry/utils/imports.py", line 45, in import_submodules
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/local/lib/python2.7/site-packages/sentry/models/debugfile.py", line 20, in <module>
    from sentry.models.file import File, clear_cached_files
  File "/usr/local/lib/python2.7/site-packages/sentry/models/file.py", line 22, in <module>
    from sentry.app import locks
  File "/usr/local/lib/python2.7/site-packages/sentry/app.py", line 30, in <module>
    locks = LockManager(RedisLockBackend(redis.clusters.get("default")))
  File "/usr/local/lib/python2.7/site-packages/sentry/utils/redis.py", line 158, in get
    raise KeyError(u"Invalid cluster type, expected: {}".format(self.__cluster_type))
KeyError: u'Invalid cluster type, expected: Redis Blaster Cluster'
```

## Steps to Reproduce

1. Configure `config.yml` as below in kubernetes using https://github.com/sentry-kubernetes/charts
```yml
################
#    Redis     #
################
redis.clusters:
  default:
    is_redis_cluster: true
    hosts:
      0:
        host: "sentry-redis-cluster.monitoring.svc"
        port: 6379
        password: ""
```
    - This didn't work
2. In addition to step 1, configure `SENTRY_CACHE = "sentry.cache.redis.RedisClusterCache"` in `sentry.conf.py`
    - This also doesn't work with same error log
3. and so on.

Good items to include here include:

- Include a stacktrace or other logs when relevant
- Include a redacted version of your configuration (`sentry.conf.py` and `config.yml` files) when relevant

### What you expected to happen

Sentry uses redis cluster properly

### Possible Solution

Maybe https://github.com/getsentry/sentry/blob/caf1c5b2a54d389ebcfd57142aaa9c556622c7b2/src/sentry/utils/redis.py#L170 should be renamed to `clusters` as comment says

**Comment by kimxogus** [comment]
It seems to be related to #14818

**Comment by BYK** [comment]
Closing as #14818 is also closed.

**Comment by mattrobenolt** [comment]
See my comment here: https://github.com/getsentry/sentry/issues/14818#issuecomment-716531225

**Comment by BYK** [comment]
Will still keep this one closed in favor of #148181
